### ISSUE TITLE
[@remix-run/cloudflare-pages] `ERESOLVE overriding peer dependency` when creating a new Remix app

### ISSUE THREAD

**Issue by iamyuu** [remix-run/remix#2504]
### What version of Remix are you using?

1.3.3

### Steps to Reproduce

just creating new apps and using Cloudflare pages for deployment target

```
npx create-remix@latest
? Where would you like to create your app? ./my-remix-app
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure, it's easy to change deployment targets. Cloudflare Pages
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
```

### Expected Behavior

it should not show warn overriding peer dependency

### Actual Behavior

<details>
<summary><code>npx create-remix@latest</code></summary>
<pre>
? Where would you like to create your app? ./my-remix-app
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure, it's easy to change deployment targets. Cloudflare Pages
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @remix-run/cloudflare@1.3.3
npm WARN Found: @cloudflare/workers-types@3.4.0
npm WARN node_modules/@cloudflare/workers-types
npm WARN   dev @cloudflare/workers-types@"^3.2.0" from the root project
npm WARN   1 more (@remix-run/cloudflare-pages)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer @cloudflare/workers-types@"^2.2.2" from @remix-run/cloudflare@1.3.3
npm WARN node_modules/@remix-run/cloudflare
npm WARN   @remix-run/cloudflare@"1.3.3" from @remix-run/cloudflare-pages@1.3.3
npm WARN   node_modules/@remix-run/cloudflare-pages
npm WARN
npm WARN Conflicting peer dependency: @cloudflare/workers-types@2.2.2
npm WARN node_modules/@cloudflare/workers-types
npm WARN   peer @cloudflare/workers-types@"^2.2.2" from @remix-run/cloudflare@1.3.3
npm WARN   node_modules/@remix-run/cloudflare
npm WARN     @remix-run/cloudflare@"1.3.3" from @remix-run/cloudflare-pages@1.3.3
npm WARN     node_modules/@remix-run/cloudflare-pages
npm WARN deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.
</pre>
</details>

**Comment by iamyuu** [comment]
when changing dependencies (e.g. exact the version) and then running `npm install`, it's throwing an error:

```
npm install

> postinstall
> remix setup cloudflare

Error: Cannot find module '@remix-run/cloudflare/package.json'
Require stack:
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/setup.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/commands.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/run.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/index.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/index.js
- /Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.resolve (node:internal/modules/cjs/helpers:108:19)
    at resolvePackageJsonFile (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/setup.js:111:18)
    at Object.setupRemix (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/setup.js:65:29)
    at Object.setup (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/commands.js:132:17)
    at Object.run (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/run.js:360:22)
    at Object.<anonymous> (/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli.js:16:11)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/setup.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/commands.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/run.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli/index.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/index.js',
    '/Users/iamyuu/dev/sandbox/my-remix-app/node_modules/@remix-run/dev/cli.js'
  ]
}
npm ERR! code 1
npm ERR! path /Users/iamyuu/dev/sandbox/my-remix-app
npm ERR! command failed
npm ERR! command sh -c remix setup cloudflare
```

**Comment by Nipsuli** [comment]
Run into same. Versions I'm running:

```
❯ npm show create-remix version
1.3.4

❯ node --version
v16.13.1

❯ npm --version
8.1.2

❯ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal
```

Trying to create type script remix app to be deployed to Cloudflare Pages:
```
❯ npx create-remix@latest
? Where would you like to create your app? ./my-remix-app
? What type of app do you want to create? Just the basics
? Where do you want to deploy? Choose Remix if you're unsure, it's easy to change deployment targets. Cloudflare Pages
? TypeScript or JavaScript? TypeScript
? Do you want me to run `npm install`? Yes
```

Fails with:
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: remix-template-cloudflare-pages@undefined
npm ERR! Found: @cloudflare/workers-types@3.4.0
npm ERR! node_modules/@cloudflare/workers-types
npm ERR!   dev @cloudflare/workers-types@"^3.2.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @cloudflare/workers-types@"^2.2.2" from @remix-run/cloudflare@1.3.4
npm ERR! node_modules/@remix-run/cloudflare
npm ERR!   @remix-run/cloudflare@"^1.3.4" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /home/nipsuli/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/nipsuli/.npm/_logs/2022-04-05T08_02_23_578Z-debug.log
```

Workaround: running `npm install --force` manually

**Comment by MichaelDeBoey** [comment]
This will be fixed in the upcoming release since #2530 is merged
